ddtrace/sampling_rule.py

```diff
@@ -1,244 +1,10 @@
-                    else:
-                        value = int(value)
-
-                tag_match = self._tag_value_matchers[tag_key].match(str(value))
-            else:
-                continue
-            # if we don't match with all specified tags for a rule, it's not a match
-            if tag_match is False:
-                return False
-
-        return tag_match
-
-    def sample(self, span):
-        """
-        Return if this rule chooses to sample the span
-
-        :param span: The span to sample against
-        :type span: :class:`ddtrace._trace.span.Span`
-        :returns: Whether this span was sampled
-        :rtype: :obj:`bool`
-        """
-        if self.sample_rate == 1:
-            return True
-        elif self.sample_rate == 0:
-            return False
-
-        return ((span._trace_id_64bits * KNUTH_FACTOR) % _MAX_UINT_64BITS) <= self._sampling_id_threshold
-
-    def _no_rule_or_self(self, val):
-        if val is self.NO_RULE:
-            return "NO_RULE"
-        elif val is None:
-            return "None"
-        elif type(val) == GlobMatcher:
-            return val.pattern
-        else:
-            return val
-
-    def choose_matcher(self, prop):
-        # We currently support the ability to pass in a function, a regular expression, or a string
-        # If a string is passed in we create a GlobMatcher to handle the matching
-        if callable(prop) or isinstance(prop, pattern_type):
-            # deprecated: passing a function or a regular expression'
-            deprecate(
-                "Using methods or regular expressions for SamplingRule matching is deprecated. ",
-                message="Please move to passing in a string for Glob matching.",
-                removal_version="3.0.0",
-                category=DDTraceDeprecationWarning,
-            )
-            return prop
-        # Name and Resource will never be None, but service can be, since we str()
-        #  whatever we pass into the GlobMatcher, we can just use its matching
-        elif prop is None:
-            prop = "None"
-        else:
-            return GlobMatcher(prop) if prop != SamplingRule.NO_RULE else SamplingRule.NO_RULE
-
-    def __repr__(self):
-        return "{}(sample_rate={!r}, service={!r}, name={!r}, resource={!r}, tags={!r}, provenance={!r})".format(
-            self.__class__.__name__,
-            self.sample_rate,
-            self._no_rule_or_self(self.service),
-            self._no_rule_or_self(self.name),
-            self._no_rule_or_self(self.resource),
-            self._no_rule_or_self(self.tags),
-            self.provenance,
-        )
-
-    __str__ = __repr__
-
-    def __eq__(self, other: Any) -> bool:
-        if not isinstance(other, SamplingRule):
-            return False
-        return str(self) == str(other)
+deprecate(
+    "The ddtrace.sample_rule module is deprecated and will be removed.",
+    message="Use ddtrace configurations to set sampling rules.",
```

Comment:
Should we be a bit more specific here? Like encourage the use of `DD_TRACE_SAMPLING_RULES` instead?
```suggestion
    message="Use DD_TRACE_SAMPLING_RULES to set sampling rules.",
```